### DISCUSSION TITLE
BaseText type referenced in logging/HUDController?

### DISCUSSION THREAD

**Discussion by ghost** [gnembon/fabric-carpet#640] [Q&A]
Hi, I was looking at some of the mod's code and came across the BaseText type... however I could not find a definition for it. Does anyone know what it does?
![Screen Shot 2020-12-26 at 2 54 13 PM](https://user-images.githubusercontent.com/64120270/103145201-51e5b680-478a-11eb-84fa-1bafa5f632e2.png)

**Comment by altrisi** [comment]
BaseText is a Minecraft class which is basically text, but it can have styling (color, format), or, if `TranslatableText`, translations.

In this case it is used by Carpet in order to return a Text with format (`Messenger.c(...)` processes format from everything before the first space, with the following format:

https://github.com/gnembon/fabric-carpet/blob/ba2586743aa6c97b4fda88cbadf27216c320370e/src/main/java/carpet/utils/Messenger.java#L34-L66
).

**Reply by altrisi** [reply]
Did this answer your question?
